### PR TITLE
[cub] update to 2.1.0

### PR DESCRIPTION
**Pull request by jiayuehua**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [ ] ~Any patches that are no longer applied are deleted from the port's directory.~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/cub/fix-usage.patch [hunk 1]
```diff
@@ -1,24 +1,12 @@
...
-diff --git a/cub/cmake/cub-header-search.cmake.in b/cub/cmake/cub-header-search.cmake.in
-index 2a46606..d2c38ef 100644
---- a/cub/cmake/cub-header-search.cmake.in
-+++ b/cub/cmake/cub-header-search.cmake.in
-@@ -3,6 +3,6 @@ unset(_CUB_VERSION_INCLUDE_DIR CACHE) # Clear old result to force search
- find_path(_CUB_VERSION_INCLUDE_DIR cub/version.cuh
-   NO_DEFAULT_PATH # Only search explicit paths below:
-   PATHS
--    "${CMAKE_CURRENT_LIST_DIR}/../../../@CMAKE_INSTALL_INCLUDEDIR@" # Install tree
-+    "${CMAKE_CURRENT_LIST_DIR}/../../@CMAKE_INSTALL_INCLUDEDIR@" # Install tree
```
Comment (BillyONeal) [inline ports/cub/fix-usage.patch:22]: @MarkIanHolland @allisonvacanti Do you agree this part of the patch is no longer necessary given https://github.com/NVIDIA/cub/pull/452/ ?

Comment (allisonvacanti) [reply ports/cub/fix-usage.patch:22]: This is correct -- but be aware that the CUB codebase has moved to the [NVIDIA/cccl repo](https://github.com/nvidia/cccl), and the old NVIDIA/cub, NVIDIA/thrust, and NVIDIA/libcudacxx repositories are no longer up-to-date and will be archived soon.

Comment (BillyONeal) [reply ports/cub/fix-usage.patch:22]: Sure, but that's going to be after this version. Will take this then, thanks!

**Review by BillyONeal — APPROVED** [review]
Thanks for the update!

**Pull request merged**